utils.go

```diff
@@ -156,6 +156,34 @@ func (c *testConn) SetDeadline(t time.Time) error      { return nil }
 func (c *testConn) SetReadDeadline(t time.Time) error  { return nil }
 func (c *testConn) SetWriteDeadline(t time.Time) error { return nil }
 
+// Copied from:
```

Comment:
"copied from" sounds kinda bad. I'd replace it with something like "based on" or "inspired by" or maybe even "adapted from"

Reply:
I am sorry that I am not good at English, so there may be some words that are not appropriate.

And these codes are not tested.

Reply:
Could you pplease change it to:
`Adapted from https://github.com/jshttp/fresh`